doc/source/timeseries.rst

```diff
@@ -557,6 +557,50 @@ We are stopping on the included end-point as it is part of the index
    dft2 = dft2.swaplevel(0, 1).sort_index()
    dft2.loc[idx[:, '2013-01-05'], :]
 
+.. versionadded:: 0.21.1
+
+``DatetimeIndex`` partial string indexing can be used with naive datetime-like labels when the ``DatetimeIndex`` has no timezone set.
+If a timezone is provided by the label, the datetime index is assumed to be UTC and a ``UserWarning`` is emitted.
+
+.. note::
+
+   This both works with ``pd.Timestamp`` and strings
+
+.. ipython:: python
+   :okwarning:
+
+   first_january_implicit_utc = pd.date_range('2016-01-01T00:00', '2016-01-01T23:59',
```

Comment:
It can be shortened but I would keep it a bit longer than the first 10 because of the comparison in the end.

Reply:
Which comparison?

Reply:
I thought (without carefully checking) that maybe in the end I will just compare two empty dataframes which will accidentially happen to be equal. To avoid such wrong positive test I thought having a bit longer df can be helpful.

Reply:
This are the docs, not tests. And you perfectly control what you do in the example, so you can just make it a bit longer than needed for the slicing to see the effect.